### PR TITLE
Deliver bug fixes

### PR DESCRIPTION
**Pull request by mcmajam**
### Changelog category (leave one):
- Bug Fix (user-visible misbehavior in official stable or prestable release)

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
...

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 1]
```diff
@@ -71,23 +71,33 @@ bool DatetimeAdd::convertImpl(String & out, IParser::Pos & pos)
...
+        out = std::format("toDateTime64(plus(toDecimal128({}, 9), toDecimal128({}/1000, 9)), 9)", datetime, offset);
+    }
+    else if (period == "MICROSECOND")
+    {
+        out = std::format("toDateTime64(plus(toDecimal128({}, 9), toDecimal128({}/1000000, 9)), 9)", datetime, offset);
+    }
+    else if (period == "NANOSECOND")
+    {
+        out = std::format("toDateTime64(plus(toDecimal128({}, 9), toDecimal128({}/1000000000, 9)), 9)", datetime, offset);
+    }
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:99]: I'd probably seek to implement this in the `date_add` function instead.

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 2]
```diff
@@ -144,11 +172,29 @@ bool DatetimeDiff::convertImpl(String & out, IParser::Pos & pos)
...
+        out = std::format("toInt64(multiply(minus(toDecimal128({}, 3), toDecimal128({}, 3)), 1000))", datetime_lhs, datetime_rhs);
+    }
+    else if (period == "MICROSECOND")
+    {
+        out = std::format("toInt64(multiply(minus(toDecimal128({}, 6), toDecimal128({}, 6)), 1000000))", datetime_lhs, datetime_rhs);
+    }
+    else if (period == "NANOSECOND")
+    {
+        out = std::format("toInt64(multiply(minus(toDecimal128({}, 9), toDecimal128({}, 9)), 1000000000))", datetime_lhs, datetime_rhs);
+    }
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:197]: I'd probably seek to implement this in the `dateDiff` function instead.

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp [hunk 3]
```diff
@@ -102,16 +112,18 @@ bool DatetimePart::convertImpl(String & out, IParser::Pos & pos)
...
     if (pos->type == TokenType::Comma)
     {
-         ++pos;
-         date = getConvertedArgument(fn_name, pos);
+        ++pos;
+        date = getConvertedArgument(fn_name, pos);
     }
     String format;
+    String head = "";
+    String trail = "";
```
Comment (ltrk2) [inline src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:126]: You could just use their default constructors and it'd amount to the same thing:
```suggestion
    String head;
    String trail;
```

**Review by ltrk2 — CHANGES_REQUESTED** [review]

File: src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp (hunk 2 above)
Comment (mcmajam) [reply src/Parsers/Kusto/KustoFunctions/KQLDateTimeFunctions.cpp:185]: date diff in ClickHouse does not support fractional seconds. It is all based on seconds and the underlying transforms/timezone conversions.

**Review by ltrk2 — APPROVED** [review]

**Pull request merged**